amazon/ion/ioncmodule.c

```diff
@@ -458,16 +460,80 @@ static iERR ionc_write_struct(hWRITER writer, PyObject* map, PyObject* tuple_as_
  */
 static iERR ionc_write_big_int(hWRITER writer, PyObject *obj) {
     iENTER;
-    PyObject* int_str = PyObject_CallMethod(obj, "__str__", NULL);
-    ION_STRING string_value;
-    ion_string_from_py(int_str, &string_value);
-    ION_INT ion_int_value;
 
+    PyObject* ion_int_base = PyLong_FromLong(II_MASK + 1);
+    PyObject* temp = Py_BuildValue("O", obj);
+    PyObject * pow_value, *size, *res, *py_digit, *py_reminder = NULL;
+
+    ION_INT ion_int_value;
     IONCHECK(ion_int_init(&ion_int_value, NULL));
-    IONCHECK(ion_int_from_string(&ion_int_value, &string_value));
+
+    // Determine sign
+    if (PyObject_RichCompareBool(temp, PyLong_FromLong(0), Py_LT) == 1) {
+        ion_int_value._signum = -1;
+        temp = PyNumber_Negative(temp);
+    } else if (PyObject_RichCompareBool(temp, PyLong_FromLong(0), Py_GT) == 1) {
```

Comment:
Should `PyLong_FromLong(0)` be stored in a variable to avoid recomputing it? Or is 0 a special case that requires no computation?

Reply:
Changed.